### ISSUE TITLE
TypeError: startswith first arg must be a str or tuple of str, not PurePosixPath

### ISSUE THREAD

**Issue by sha-trix** [pyinstaller/pyinstaller#6018]
### Context information (for bug reports)

* Output of `pyinstaller --version`: ```4.4```
* Version of Python: 3.6.7
* Platform: Ubuntu
* How you installed Python: container
* Did you also try this on another platform? Does it work there?
  No

* try the latest development version, using the following command:

```shell
pip install https://github.com/pyinstaller/pyinstaller/archive/develop.zip
```

* follow *all* the instructions in our "If Things Go Wrong" Guide
  (https://github.com/pyinstaller/pyinstaller/wiki/If-Things-Go-Wrong) and

### Make sure [everything is packaged correctly](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs#make-sure-everything-is-packaged-correctly)

  * [ ] start with clean installation
  * [ ] use the latest development version
  * [ ] Run your frozen program **from a command window (shell)** — instead of double-clicking on it
  * [ ] Package your program in **--onedir mode**
  * [ ] Package **without UPX**, say: use the option `--noupx` or set `upx=False` in your .spec-file
  * [ ] Repackage you application in **verbose/debug mode**. For this, pass the option `--debug` to `pyi-makespec` or `pyinstaller` or use `EXE(..., debug=1, ...)` in your .spec file.

### A minimal example program which shows the error

```
```docker
pyinstaller  --hidden-import saml2.attributemaps.adfs_v1x --hidden-import saml2.attributemaps.adfs_v20 --hidden-import saml2.attributemaps.basic --hidden-import saml2.attributemaps.saml_uri --hidden-import saml2.attributemaps.shibboleth_uri -y -D  Test.py
``````

### Stacktrace / full error message

![image](https://user-images.githubusercontent.com/30793313/125938547-319e3db2-9678-4250-9262-381bc06edbb8.png)

```
(paste text here)
```

Please also see <https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs>
for more about what would use to solve the issue.

**Comment by rokm** [comment]
Thanks for the report! 

Can you check if using the branch from #6019 fixes the issue? (You can install that version of PyInstaller via `pip install https://github.com/rokm/pyinstaller/archive/refs/heads/fix-6018.zip`).

**Comment by sha-trix** [comment]
> Thanks for the report!
> 
> Can you check if using the branch from #6019 fixes the issue? (You can install that version of PyInstaller via `pip install https://github.com/rokm/pyinstaller/archive/refs/heads/fix-6018.zip`).

that worked for me

**Comment by rokm** [comment]
Fixed in `develop` by #6018.

**Comment by pavanmajji03** [comment]
Great !!! That fixes the issue.  Thx for the update!!!

**Comment by lstolcman** [comment]
example error message regarding this type of bug, in text format (it took me a while to find a fix, #6019, now it should be easier for other people, since google can handle that):
```
Traceback (most recent call last):
  File "PrngCipher.py", line 2, in <module>
    from kivy.app import App
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller\loader\pyimod03_importers.py", line 546, in exec_module
  File "kivy\__init__.py", line 272, in <module>
  File "C:\Program Files\Python39\Lib\site-packages\PyInstaller\hooks\rthooks\pyi_rth_pkgutil.py", line 71, in _pyi_pkgutil_iter_modules
    assert pkg_path.startswith(SYS_PREFIX)
TypeError: startswith first arg must be str or a tuple of str, not PureWindowsPath
```
